aws/resource_aws_connect_instance_test.go

```diff
@@ -0,0 +1,346 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/connect"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_connect_instance", &resource.Sweeper{
+		Name: "aws_connect_instance",
+		F:    testSweepConnectInstance,
+	})
+}
+
+func testSweepConnectInstance(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).connectconn
+	ctx := context.Background()
+	// MaxResults:  Maximum value of 10. https://docs.aws.amazon.com/connect/latest/APIReference/API_ListInstances.html
+	input := &connect.ListInstancesInput{MaxResults: aws.Int64(10)}
+	var sweeperErrs *multierror.Error
+	for {
+		listOutput, err := conn.ListInstances(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping Connect Instance sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving Connect Instance: %s", err)
+		}
+		for _, instance := range listOutput.InstanceSummaryList {
+			id := aws.StringValue(instance.Id)
+			r := resourceAwsConnectInstance()
+			d := r.Data(nil)
+			d.SetId(id)
+
+			diags := r.DeleteContext(ctx, d, client)
+			for i := range diags {
+				if diags[i].Severity == diag.Error {
+					log.Printf("[ERROR] %s", diags[i].Summary)
+					sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf(diags[i].Summary))
+					continue
+				}
+			}
+		}
+		if aws.StringValue(listOutput.NextToken) == "" {
+			break
+		}
+		input.NextToken = listOutput.NextToken
+	}
+	return sweeperErrs.ErrorOrNil()
+}
+func TestAccAwsConnectInstance_basic(t *testing.T) {
+	var v connect.DescribeInstanceOutput
+	rInt := acctest.RandInt()
+	resourceName := "aws_connect_instance.foo"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsConnectInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsConnectInstanceConfigBasic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsConnectInstanceExists(resourceName, &v),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
```

Comment:
see https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/running-and-writing-acceptance-tests.md#hardcoded-partition-in-arn